aspnet/testing/unit-testing.rst

```diff
@@ -2,28 +2,131 @@ Unit Testing
+	src/
+	   PrimeWeb/
+	      project.json
+		  Startup.cs
+		  Services/
+		     PrimeService.cs
+	test/
+	   PrimeWeb.UnitTests/
+	      project.json
+		  Services/
+		     PrimeService_IsPrimeShould.cs
+		  
+It is important that there be a folder/directory with the name of the project you are testing (PrimeWeb above), since the file system is used to find your project.
+
+Configuring the Test project.json
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The test project's ``project.json`` file should add dependencies on the test framework being used and the SUT project. For example, to work with the `xUnit test framework <http://xunit.github.io/>`_, you would configure the dependencies as follows:
+
+.. literalinclude:: unit-testing/sample/test/PrimeWeb.UnitTests/project.json
+	:language: javascript
+	:lines: 20-24
+	:linenos:
+	:dedent: 2
+
+As other test frameworks release support for DNX, we will link to them here. We are simply using xUnit as one example of the many different testing frameworks that can be plugged into ASP.NET and DNX.
+
+.. note:: Be sure the version numbers match for your project-to-project references.
+
+In addition to adding dependencies, we also want to be able to run the tests using a dnx command. To do so, add the following commands section to ``project.json``:
 
+.. literalinclude:: unit-testing/sample/test/PrimeWeb.UnitTests/project.json
+	:language: javascript
+	:lines: 25-27
+	:linenos:
+	:dedent: 2
+
+.. note:: Learn more about :doc:`/dnx/commands` in DNX.
+	
 Running Tests
 -------------
-Running unit tests on the command line and in visual studio
+Before you can run your tests, you'll need to write some. For this demo, I've created a simple service that checks whether numbers are prime. One of the tests is shown here:
+
+.. literalinclude:: unit-testing/sample/test/PrimeWeb.UnitTests/Services/PrimeService_IsPrimeShould.cs
+	:language: c#
+	:lines: 18-27
+	:linenos:
+	:dedent: 8
+
+This test will check the values -1, 0, and 1 using the ``IsPrime`` method in each of three separate tests. Each test will pass if ``IsPrime`` returns false, and will otherwise fail.
+
+You can run tests from the command line or using Visual Studio, whichever you prefer.
+
+Visual Studio
+^^^^^^^^^^^^^
+To run tests in Visual Studio, first open the Test Explorer tab, then build the solution to have it discover all available tests. Once you have done so, you should see all of your tests in the Test Explorer window. Click Run All to run the tests and see the results.
+
+.. image:: unit-testing/_static/test-explorer.png
+
+If you click the icon in the top-left, Visual Studio will run tests after every build, providing immediate feedback as you work on your application.
+
+Command Line
+^^^^^^^^^^^^
+To run tests from the command line, navigate to your unit test project folder. Next, run
+
+.. code-block:: powershell
+
+	dnx test
+
+You should see output similar to the following:
+
+.. image:: unit-testing/_static/dnx-test.png
+
+dnx-watch
```

Comment:
Assuming that tests have a direct reference to the project being tested then it is a perfectly valid scenario